lib/document.js

```diff
@@ -533,6 +553,13 @@ class Document extends Readable {
       await this._writeObject(destsObj)
       catalog.prop('Names', destsObj.toReference())
     }
+    //Write outlines hierarchy if there are outlines
+    if(this._outlines.length > 1) {
+      for (let i = 0; i < this._outlines.length; i += 1) {
+        await this._writeObject(this._outlines[i])
+      }
+      catalog.prop('Outlines', this._outlines[0].toReference())  
```

Comment:
You could call `this._registerObject(this._outlines[0])` here (as mentioned in the previous comment)

Reply:
see previous comment